compiler/rustc_interface/src/interface.rs

```diff
@@ -280,6 +280,8 @@ pub fn run_compiler<R: Send>(config: Config, f: impl FnOnce(&Compiler) -> R + Se
                 registry.clone(),
             );
 
+            rustc_data_structures::sync::set_parallel(sess.threads() > 1);
```

Comment:
This probably needs to move up before thread pool creation as the session will already create locks.

Reply:
Yea, that makes sense, thanks!